framework/src/meshgenerators/DistributedRectilinearMeshGenerator.C

```diff
@@ -840,6 +849,12 @@ DistributedRectilinearMeshGenerator::buildCube(UnstructuredMesh & mesh,
   // Current processor ID
   const auto pid = comm.rank();
 
+  if (_num_cores_for_partition > num_procs)
+    mooseError("Number of cores for partition is too large ", _num_cores_for_partition);
+
+  if (_num_cores_for_partition <= 0)
```

Comment:
Both dof_id_type and processor_id_type are unsigned so I'd be very scared if we ever had a number less than zero!

Reply:
:smile:

Reply:
How could I forget that?!

Reply:
Done